src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,50 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size = 0;
+    private T[] elements;
 
+    public MateStack() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void push(T value) {
+        if (size == elements.length) {
+            elements = resize();
+        }
+        elements[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        checkIfIsEmpty();
+        return elements[size - 1];
     }
 
     public T pop() {
-        return null;
+        checkIfIsEmpty();
+        T popElement = elements[size - 1];
+        elements[size - 1] = null;
+        size--;
```

Comment:
you can simplify the code
```suggestion
        elements[--size] = null;
```

Reply:
agree